cluster/node.go

```diff
@@ -153,32 +154,33 @@ func (n *Node) refreshImages() error {
 	return nil
 }
 
-// Refresh the list and status of containers running on the node.
-func (n *Node) refreshContainers() error {
+// Refresh the list and status of containers running on the node. If `full` is
+// true, each container will be inspected.
+func (n *Node) RefreshContainers(full bool) error {
 	containers, err := n.client.ListContainers(true, false, "")
 	if err != nil {
 		return err
 	}
 
-	n.Lock()
-	defer n.Unlock()
-
 	merged := make(map[string]*Container)
 	for _, c := range containers {
-		merged, err = n.updateContainer(c, merged)
+		merged, err = n.updateContainer(c, merged, full)
```

Comment:
`n.updateContainer` is using `n.containers` I think the lock is required before, no ?